components/script/dom/htmlformelement.rs

```diff
@@ -395,6 +397,28 @@ impl HTMLFormElementMethods for HTMLFormElement {
             }
         }
 
+        // Step 4
+        let past_names_map = self.past_names_map.borrow();
+        for (key, val) in past_names_map.iter() {
+            let entry = SourcedName {
+                name: key.clone(),
+                element: DomRoot::from_ref(&*val.0),
+                source: SourcedNameSource::Past(now()-val.1), // calculate difference now()-val.1 to find age
+            };
+            sourcedNamesVec.push(entry);
+        }
+
+        // Step 5
+        // TODO need to sort as per spec. This is a partially implemented function.
+        // Kindly guide us on how to refine this function further.
+        sourcedNamesVec.sort_by(|a, b| a.name.partial_cmp(&b.name).unwrap());
```

Comment:
@jdm As suggested in the [project page](https://github.com/servo/servo/wiki/Missing-DOM-features-project) we tried to use `Node::CompareDocumentPosition`. But since its return type is `u16` and not `std::cmp::Ordering` we were getting error while trying to implement Step 5 [SupportedPropertyNames](https://html.spec.whatwg.org/multipage/forms.html#the-form-element:supported-property-names) as follows:

sourcedNamesVec.sort_by(|a, b| a.element.upcast::<Node>().CompareDocumentPosition(b.element.upcast::<Node>()));

Kindly help us how to sort `sourcedNamesVec` as mentioned in the spec.